crates/starknet/src/starknet/add_deploy_account_transaction.rs

```diff
@@ -121,10 +121,47 @@ mod tests {
         let (txn_hash, _) = starknet.add_deploy_account_transaction(transaction).unwrap();
         let txn = starknet.transactions.get_by_hash_mut(&txn_hash).unwrap();
 
+        assert_eq!(txn.finality_status, None);
+        assert!(txn.execution_result.revert_reason().unwrap().contains("Fee transfer failure"));
+    }
+
+    #[test]
+    fn deploy_account_transaction_should_fail_due_to_not_enough_fee() {
```

Comment:
What is the difference between this and the previous test (`deploy_account_transaction_should_fail_due_to_low_balance`)?

Reply:
In this test the account that is making the transaction have some balance in fee_token_contract, but due to the low max fee that it provided, the transaction fails and the max fee is being charged.
In the other one the account doesn't have any balance, so when it tries to charge with the amount = max_fee, it returns an error, something like an exception from starknet_in_rust